overlord/snapstate/backend/link.go

```diff
@@ -72,6 +72,10 @@ func (b Backend) LinkSnap(info *snap.Info) error {
 }
 
 func generateWrappers(s *snap.Info) error {
+	// add the environment
+	if err := wrappers.AddSnapEnvironment(s); err != nil {
```

Comment:
This has to be influenced by interfaces. I wonder if it would make sense to make this the same as all the other content created by interfaces. I don't mean that it has to happen now but we're pretty sure it has to happen.